vm/string.go

```diff
@@ -297,6 +299,36 @@ var builtinStringMethods = []*BuiltInMethod{
 		},
 		Name: "to_s",
 	},
+	{
+		Fn: func(receiver Object) builtinMethodBody {
+			return func(vm *VM, args []Object, blockFrame *callFrame) Object {
+
+				str := receiver.(*StringObject).Value
+				parsedStr, err := strconv.ParseInt(str, 10, 0)
+
+				if err == nil {
+					return initilaizeInteger(int(parsedStr))
+				}
+
+				var digits string
```

Comment:
Sorry, not implementation, I mean the outcome. Go's `strconv.ParseInt` returns 0 if parsing strings that contain any non-digit character like `"123string"`. 

However, Ruby's `String#to_i` method goes through the string and returns all digits before the first a non-digit character. For example, `"123string".to_i` would return `"123"`, and `"str123"` would return `0`.